### DISCUSSION TITLE
New Quiz Items

### DISCUSSION THREAD

**Discussion by jamiehomewood** [ucfopen/canvasapi#632] [Q&A] [answered]
Hi All,

Is it possible to get the new quiz items yet? and if, where are the methods for this?

Cheers

J

**Comment by bennettscience** [comment, marked as answer]
The New Quiz API [does allow access to items](https://canvas.instructure.com/doc/api/new_quiz_items.html), but the methods are not active in the library yet. #609 is open as an issue to add the endpoints if you want to take a shot at adding it.

In the meantime, you can use the `Requester` module to issue a custom request to get items for a new quiz.

**Reply by marty-sullivan** [reply]
@bennettscience is there an example of using `Requester` to do this?

**Reply by bennettscience** [reply]
The `Requester` API has a page in the [docs](https://canvasapi.readthedocs.io/en/stable/requester-ref.html) you can read over, but it's something along these lines (not tested):

```python
from canvasapi import Canvas
from canvasapi.utils import combine_kwargs

# The Canvas object has an initialized requester attached that you can access.
# See more 
# It will pass your credentials along on the request.
canvas = Canvas(URL, KEY)

# Assuming these come from some other request...
course_id = 1
assignment_id = 12345

# Any parameters you want to pass to the request
opts = {}


resp = canvas.__requester.request(
    "GET",
    "/api/quiz/v1/courses/:course_id/quizzes/{}/items/{}".format(course_id, assignment_id),
    _kwargs=combine_kwargs(opts)
)

# do more with the response...
```

One thing to note is that if you're building a request on an existing `CanvasObject` like an `Assignment`, the requester is a single-leading underscore. If it's on the main `Canvas` object, it is a double underscore.

**Reply by marty-sullivan** [reply]
I needed to do something like this, but it works:

```python
@classmethod
def delete_new_quiz(cls, course_id: int, quiz_id: int) -> None:
    requester = Requester(cls.canvas_api_url, cls.canvas_api_key)

    requester.request(
        'DELETE',
        _url=f'{cls.canvas_api_url}/api/quiz/v1/courses/{course_id}/quizzes/{quiz_id}',
    )
```
